### ISSUE TITLE
please drop dependency on common-path

### ISSUE THREAD

**Issue by WeepingClown13** [zellij-org/zellij#3702]
A bit of context first; I am packaging zellij for Debian, for a while now. There are tons of dependencies that had to be and has to be packaged first, as per Debian policy. Currently I have finished packaging all dependencies/dev-dependencies of zellij-utils, except [common-path](https://crates.io/crates/common-path).

`common-path` is currently version 1.0.0, and the last update to it was 6 years ago. That much is quite common, but this one also points to an nonexistent repository as source. Examining the [source code](https://docs.rs/crate/common-path/1.0.0/source/src/lib.rs), there is only so much - non complex - logic existing there. with less than 50 lines of actual library code excluding tests. At that point, having it as a separate package becomes only a lot of maintenance burden. Examining the common-path usage in zellij, it only seems to be used at a single place in zellij-utils.

In conclusion, it'd make life a lot of easier for downstream packaging if this dependency can be avoided. And that leaves some questions.

1. Can the usage of common-path itself easily be removed? Would that affect anything severely?
2. Are there better or well maintained alternatives that could be used, if the utility it provides is a must have?
3. Even better, is it possible to zellij itself to inline its own logic for this so that the utility is retrieved and it's one less dependency?

Please share your thoughts!

**Comment by imsnif** [comment]
Hey, thank you for your efforts! I would be very happy to see Zellij packaged for Debian. common-path is only used in one place in the code base: https://github.com/zellij-org/zellij/blob/main/zellij-server/src/session_layout_metadata.rs#L252 

We use it there to find the common path of terminal working directories when serializing a layout for the session resurrection. This code isn't overly complex but can potentially be error prone. I'm happy to accept a PR that does this manually instead using the STD lib's primitives. I guess it shouldn't be overly complex. Wdyt?

**Comment by imsnif** [comment]
My apologies - I was a bit unclear. I would be happy to use the STD lib's primitives. :)

Right now I'm mostly concerned with getting the next release out the door, but once that's done - if you haven't gotten this done yet I'll try to whip something up. This honestly sounds like a 5-10 lines of code function to me, but I'm always open to having missed something.

**Comment by joshuamegnauth54** [comment]
You may consider vendoring this dependency for the next release too. Based on the code, the two functions seem easy to replicate, only use the standard library, and likely won't ever need maintenance. Plus, the original author included unit tests too. :smile_cat: 

The main logic is:
* Zip the [std::path::Component](https://doc.rust-lang.org/std/path/enum.Component.html)s of the two paths we want to test
* Iterate through the pairs until they're not equal
* The prefix is the last component where they're equal

So for example:
Path `a`: `/home/josh/.config`
Path `b`: ` /home`

**Iteration 1:**
`/` == `/`

**Iteration 2:**
`home` == `home`

**Iteration 3:**
`josh` != empty

The prefix is `/home`.
----

Anywho, thank you two for Zellij and Debian! I totally forgot why I came here now. :sweat_smile:

**Comment by imsnif** [comment]
Thanksto @s1syph0s - common_path has now been vendored! :tada:

**Comment by WeepingClown13** [comment]
Thanks @s1syph0s and @imsnif ! I originall hoped for a zellij solution without vendoring, but since it has a proper copyright file and that's embedded with the code, it doesn't really cause any problems. I guess everyone's happy :)

**Comment by WeepingClown13** [comment]
I missed the latest release because I compiled it from the source last time, looks like I can close this now. Thanks for the support.

**Comment by imsnif** [comment]
Hey @WeepingClown13 - I hope the ping is OK, I'm not sure how else to reach you. I was wondering if there is any news regarding the Debian packaging? I remember it was stuck on issues with wasmtime packaging, is that still the case? Is it possible to expedite this somehow? This remains one of the most significant barriers of entry to Zellij...

EDIT: just to be clear - I'm very much open to making changes from the Zellij side if it will help this happen. Even to the extent of switching wasm runtimes if needed.

**Comment by WeepingClown13** [comment]
@imsnif sadly it is still blocked on wasmtime. As it is a hard dependency, I have no way of moving forward for the moment until it becomes available :(

EDIT: feel free to mail me at weepingclown@debian.org btw
